src/package-hoister.js

```diff
@@ -163,6 +168,64 @@ export default class PackageHoister {
   }
 
   /**
+   * Propagate inherited ignore statuses from non-ignored to ignored packages
+  */
+
+  _propagateNonIgnored() {
+    //
+    const tovisit: Array<HoistManifest> = [];
+
+    // enumerate all non-ignored packages
+    for (const entry of this.tree.entries()) {
+      if (!entry[1].isIgnored) {
+        tovisit.push(entry[1]);
+      }
+    }
+
+    // visit them
+    while (tovisit.length) {
+      const info = tovisit.shift();
+      const ref = info.pkg._reference;
+      invariant(ref, 'expected reference');
+
+      for (const depPattern of ref.dependencies) {
+        //
+        const depinfo = this._lookupDependency(info, depPattern);
+
+        //
```

Comment:
Keeping the code density in line with the rest of the code, nothing more. No problem to remove them.